### PR TITLE
David orders

### PR DESCRIPTION
**Pull request by DavidTTran**
Completed
- Stories 23-24
- Relocate updating user information from profile_controller to user_controller
- Style profile page

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -4,33 +4,8 @@ class ProfilesController < ApplicationController
...
 
-  def existing_email?
-    User.exists?(email: params[:email]) && params[:email] != current_user.email
-  end
-
-  def user_params
-    params[:password_digest] = current_user.password_digest
-    params.permit(:name, :address, :city, :state, :zip, :email, :password_digest)
-  end
-
```
Comment (Maxwell-Baird) [inline app/controllers/profiles_controller.rb:33]: Good idea of moving these over to user_controller in order to make our application more readable.

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def update
       cart.add_quantity(params[:id]) unless cart.limit_reached?(params[:id])
     elsif params[:quantity] == "dec"
       cart.subtract_quantity(params[:id])
-      session[:cart].delete(params[:id]) if cart.quantity_zero?(params[:id])
+      return remove_item(params[:id]) if cart.quantity_zero?(params[:id])
```
Comment (Maxwell-Baird) [inline app/controllers/cart_controller.rb:30]: What is the point of this return?

File: app/controllers/cart_controller.rb [hunk 3]
```diff
@@ -17,8 +17,8 @@ def empty
     redirect_to '/cart'
   end
 
-  def remove_item
-    session[:cart].delete(params[:item_id])
+  def remove_item(item = nil)
+    item.nil? ? (session[:cart].delete(params[:item_id])) : session[:cart].delete(item)
```
Comment (colintalex) [inline app/controllers/cart_controller.rb:21]: Good use of this shorthand syntax

File: app/controllers/profiles_controller.rb (hunk 1 above)
Comment (colintalex) [reply app/controllers/profiles_controller.rb:33]: Agreed, makes more sense

File: app/models/cart.rb [hunk 4]
```diff
@@ -45,7 +45,6 @@ def quantity_zero?(item)
   end
 
   def limit_reached?(item)
-    item = Item.find(item)
-    @contents[item] == item.inventory
+    @contents[item] == Item.find(item).inventory
```
Comment (colintalex) [inline app/models/cart.rb:48]: Good AR work

**Review by colintalex — APPROVED** [review]
Looks good to me, I like the use of AR methods

File: app/controllers/cart_controller.rb (hunk 2 above)
Comment (DavidTTran) [reply app/controllers/cart_controller.rb:30]: It was producing an error when I just tried to call `remove_item(params[:id])`

**Pull request merged**